test/common/containers/DowntimeNotification.unit.spec.jsx

```diff
@@ -0,0 +1,112 @@
+import React from 'react';
+import enzyme from 'enzyme';
+import { expect } from 'chai';
+import sinon from 'sinon';
+import moment from 'moment';
+
+import { DowntimeNotification, services, serviceStatus } from '../../../src/js/common/containers/DowntimeNotification';
+
+const beforeNow = moment().subtract(1, 'minute').toDate();
+const withinHour = moment().add(1, 'hour').subtract(1, 'minute').toDate();
+const moreThanHour = moment().add(1, 'hour').add(1, 'hour').toDate();
+const endTime = moment().add(6, 'hour').toDate();
+
+const innerText = 'This is the inner text';
+function getComponent(dependencies = [], getScheduledDowntime = () => {}) {
+  return enzyme.shallow(
+    <DowntimeNotification dependencies={dependencies} scheduledDowntime={null} getScheduledDowntime={getScheduledDowntime}>
+      <span>{innerText}</span>
+    </DowntimeNotification>
+  );
+}
+
+describe('<DowntimeNotification/>', () => {
+
+  it('calls getScheduledDowntime when rendered', () => {
+    const getScheduledDowntime = sinon.spy();
+    getComponent([], getScheduledDowntime);
+    expect(getScheduledDowntime.calledOnce).to.be.true;
+  });
+
+  describe('No impending downtime', () => {
+
+    it('should render the children components when there are no dependencies', () => {
+      const wrapper = getComponent();
+      wrapper.setProps({ isReady: true, scheduledDowntime: [] });
+      expect(wrapper.text()).to.contain(innerText, 'The message was rendered.');
+    });
+
+    it('should render the children components when dependencies have no downtime', () => {
+      const wrapper = getComponent([services.mvi]);
+      wrapper.setProps({ isReady: true, scheduledDowntime: [] });
+      expect(wrapper.text()).to.contain(innerText, 'The message was rendered.');
```

Comment:
Huh. Well, if this passes, I'm satisfied. When I was trying to figure out how `getDowntimeWindow` might break things, I found if `downtimes` was an empty array, it should return `{ startTime: false, endTime: false }` which would cause an error in `renderStatusDown` or `renderStatusDownApproaching`. It also seemed like `getDowntimeWindow` _could_ be called with an empty array from `componentWillRecieveProps`.

Turns out I must be missing something, though, if this passes. :man_shrugging:

Reply:
It's confusing, but I just added a check to do the `getDowntimeWindow` only when the status isn't ok. There was already a check for that prior to this, but it wasn't clear enough. It was tough working with so many date operations.